lib/Sema/ConstraintSystem.h

```diff
@@ -2181,8 +2212,28 @@ class ConstraintSystem {
   /// \param lookThroughApply Whether to look through applies. If false, a
   /// callee locator will only be returned for a direct reference such as
   /// \c x.foo rather than \c x.foo().
+  /// \param getType The callback to fetch a type for given expression.
+  /// \param simplifyType The callback to attempt to resolve any type
+  ///                     variables which appear in the given type.
+  /// \param getOverloadFor The callback to fetch overload for a given
+  ///                       locator if available.
+  ConstraintLocator *getCalleeLocator(
+      ConstraintLocator *locator, bool lookThroughApply,
+      llvm::function_ref<Type(const Expr *)> getType,
+      llvm::function_ref<Type(Type)> simplifyType,
+      llvm::function_ref<Optional<SelectedOverload>(ConstraintLocator *)>
+          getOverloadFor);
```

Comment:
@hamishknight Please take a look at the changes for `getCalleeLocator` there is now one in `ConstraintSystem` and one in the `Solution` because ambiguity diagnostics have to use unapplied solution to retrieve type/overload information.